### PR TITLE
Update the example template for a no Trainer option

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

Expand the template of new examples with a new option to build an example like the new [run_glue_no_trainer.py](https://github.com/huggingface/transformers/blob/master/examples/text-classification/run_glue_no_trainer.py).

### CODE REVIEW

File: templates/adding_a_new_example_script/{{cookiecutter.directory_name}}/run_{{cookiecutter.example_shortcut}}.py [hunk 1]
```diff
@@ -426,3 +428,406 @@ def _mp_fn(index):
...
+
+    lr_scheduler = get_scheduler(
+        name=args.lr_scheduler_type,
+        optimizer=optimizer,
+        num_warmup_steps=args.num_warmup_steps,
+        num_training_steps=args.max_train_steps,
+    )
+
+    # TODO Get the proper metric function
+    # metric = load_metric(xxx)
```
Comment (LysandreJik) [inline templates/adding_a_new_example_script/{{cookiecutter.directory_name}}/run_{{cookiecutter.example_shortcut}}.py:779]: Is this TODO meant to be here?

**Review by LysandreJik — APPROVED** [review]
LGTM, great to have a template for this!

File: templates/adding_a_new_example_script/{{cookiecutter.directory_name}}/run_{{cookiecutter.example_shortcut}}.py (hunk 1 above)
Comment (sgugger) [reply templates/adding_a_new_example_script/{{cookiecutter.directory_name}}/run_{{cookiecutter.example_shortcut}}.py:779]: Yes I can't do it for the user, they have to implement the metric here.
